File: webook/internal/domain/interaction.go
package domain

type Interaction struct {
	BizId        int64  `json:"bizId"`
	Biz          string `json:"biz"`
	ReadCount    int64  `json:"readCount"`
	LikeCount    int64  `json:"likeCount"`
	CollectCount int64  `json:"collectCount"`
	Liked        bool   `json:"liked"`
	Collected    bool   `json:"collected"`
}

// 业务类型常量
const (
	BizArticle = "article"
	BizComment = "comment" // 评论点赞复用 interaction，biz="comment"、bizId=commentId
)

File: webook-fe/types/interaction.ts
// 对应后端 domain.Interaction
export interface Interaction {
  articleId: number;
  readCount: number;
  likeCount: number;
  collectCount: number;
  liked: boolean; // 当前用户是否已点赞
  collected: boolean; // 当前用户是否已收藏
}

// interaction 通用目标：biz 业务类型 + bizId 业务内主键（article→articleId、comment→commentId）
export interface InteractionTarget {
  biz: string;
  bizId: number;
}

// POST /interaction/like
export interface LikeReq extends InteractionTarget {
  liked: boolean; // true=点赞，false=取消
}

// POST /interaction/collect
export interface CollectReq extends InteractionTarget {
  collected: boolean; // true=收藏，false=取消
}
